demos/common/models/src/hpe_model_openpose.cpp

```diff
@@ -0,0 +1,206 @@
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include "models/hpe_model_openpose.h"
+#include "models/openpose_decoder.h"
+
+#include <samples/common.hpp>
+#include <ngraph/ngraph.hpp>
+#include <samples/ocv_common.hpp>
+
+using namespace InferenceEngine;
+
+HPEOpenPose::HPEOpenPose(const std::string& modelFileName, float confidenceThreshold) :
+    ModelBase(modelFileName),
+    confidenceThreshold(confidenceThreshold) {
+}
+
+void HPEOpenPose::prepareInputsOutputs(InferenceEngine::CNNNetwork& cnnNetwork) {
+    // --------------------------- Configure input & output -------------------------------------------------
+    // --------------------------- Prepare input blobs ------------------------------------------------------
+    ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
+    if (inputShapes.size() != 1)
+        throw std::runtime_error("Demo supports topologies only with 1 input");
+    inputsNames.push_back(inputShapes.begin()->first);
+    SizeVector& inSizeVector = inputShapes.begin()->second;
+    if (inSizeVector.size() != 4 || inSizeVector[0] != 1 || inSizeVector[1] != 3)
+        throw std::runtime_error("3-channel 4-dimensional model's input is expected");
+
+    InputInfo& inputInfo = *cnnNetwork.getInputsInfo().begin()->second;
+    inputInfo.setPrecision(Precision::U8);
+    inputInfo.getInputData()->setLayout(Layout::NCHW);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    const OutputsDataMap& outputInfo = cnnNetwork.getOutputsInfo();
+    if (outputInfo.size() != 2)
+        throw std::runtime_error("Demo supports topologies only with 2 outputs");
+
+    for (const auto& outputLayer: outputInfo) {
+        outputLayer.second->setPrecision(Precision::FP32);
+        outputLayer.second->setLayout(Layout::NCHW);
+        outputsNames.push_back(outputLayer.first);
+    }
+
+    auto outputIt = outputInfo.begin();
+    const SizeVector& pafsOutputDims = (*outputIt++).second->getTensorDesc().getDims();
+    if (pafsOutputDims.size() != 4 || pafsOutputDims[0] != 1 || pafsOutputDims[1] != 2 * (keypointsNumber + 1))
+        throw std::runtime_error("1x" + std::to_string(2 * (keypointsNumber + 1)) + "xHFMxWFM dimension of model's output is expected");
+    const SizeVector& heatmapsOutputDims = (*outputIt++).second->getTensorDesc().getDims();
+    if (heatmapsOutputDims.size() != 4 || heatmapsOutputDims[0] != 1 || heatmapsOutputDims[1] != keypointsNumber + 1)
+        throw std::runtime_error("1x" + std::to_string(keypointsNumber + 1) + "xHFMxWFM dimension of model's heatmap is expected");
+    if (pafsOutputDims[2] != heatmapsOutputDims[2] || pafsOutputDims[3] != heatmapsOutputDims[3])
+        throw std::runtime_error("output and heatmap are expected to have matching last two dimensions");
+}
+
+cv::Size HPEOpenPose::reshape(InferenceEngine::CNNNetwork& cnnNetwork, int targetSize) {
```

Comment:
I believe aspect ratio should be provided to the model instead of actual image size: currently a constructor takes two values for height: one is `targetSize`, the second is ` cv::Size::height`. That is ambiguous.

Reply:
Done.